pages/post/index.tsx

```diff
@@ -128,6 +133,47 @@ export default function NewPost() {
     }
   }, [drafts]);
 
+  // Function to handle content change in the main post
+  const handleContentChange = (event, postId) => {
+    const newPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return { ...post, content: event.target.value };
+      }
+      return post;
+    });
+    setPosts(newPosts);
+  };
+
+  // Function to add a new thread to a post
+  const addThread = (postId) => {
+    const newPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return { ...post, threads: [...post.threads, ''] };
+      }
+      return post;
+    });
+    setPosts(newPosts);
+  };
+
+  // Function to handle content change in a thread
+  const handleThreadContentChange = (event, postId, index) => {
+    const newPosts = posts.map((post) => {
+      if (post.id === postId) {
+        const newThreads = [...post.threads];
+        newThreads[index] = event.target.value;
+        return { ...post, threads: newThreads };
+      }
+      return post;
+    });
+    setPosts(newPosts);
+  };
+
+  // Function to handle post submission
+  const handlePostSubmit = () => {
+    setAllPosts([...allPosts, ...posts]);
```

Comment:
when a post is submitted, it should be sent to the Farcaster protocol. This is also done via the useDraftStore. when posting a thread, some logic needs to be applied to make it a thread: cast A can be posted as usual, but the following casts B, C, D, etc need to be replies to cast A.